pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/forward/FixedByteChunkSVForwardIndexReader.java

```diff
@@ -91,4 +95,34 @@ public double getDouble(int docId, ChunkReaderContext context) {
       return _rawData.getDouble(docId * Double.BYTES);
     }
   }
+
+  @Override
+  public List<ValueRange> getDocIdRange(int docId, ChunkReaderContext context, @Nullable List<ValueRange> ranges) {
+    if (!_isCompressed) {
+      // If uncompressed, should use fixed offset
+      throw new IllegalStateException("Operation not supported since the forward index is fixed length type");
+    }
+
+    if (ranges == null) {
+      ranges = new ArrayList<>();
+    }
+    recordDocIdRanges(docId, context, ranges);
+
+    return ranges;
+  }
+
+  @Override
+  public boolean isFixedLengthType() {
+    return !_isCompressed;
+  }
+
+  @Override
+  public long getBaseOffset() {
+    return _rawDataStart;
+  }
+
+  @Override
+  public int getDocLength() {
+    return _storedType.size();
```

Comment:
should we throw exception when `isFixedLengthType()` returns false?

Reply:
Good point. Added